frontend/src/metabase/ui/components/inputs/MultiAutocomplete/MultiAutocomplete.tsx

```diff
@@ -33,71 +36,64 @@ export function MultiAutocomplete({
     finalValue: "",
     onChange: onSearchChange,
   });
-  const [lastSelectedValues, setLastSelectedValues] = useState(selectedValues);
-  const [isFocused, setIsFocused] = useState(false);
-  const visibleValues = isFocused ? lastSelectedValues : selectedValues;
 
   const items = useMemo(
-    () => getAvailableSelectItems(data, lastSelectedValues),
-    [data, lastSelectedValues],
+    () => getAvailableSelectItems(data, selectedValues),
+    [data, selectedValues],
   );
 
   const handleChange = (newValues: string[]) => {
     setSelectedValues(newValues);
-    setLastSelectedValues(newValues);
   };
 
-  const handleFocus = (event: FocusEvent<HTMLInputElement>) => {
-    setIsFocused(true);
-    setLastSelectedValues(selectedValues);
-    onFocus?.(event);
-  };
-
-  const handleBlur = (event: FocusEvent<HTMLInputElement>) => {
-    setIsFocused(false);
-    setLastSelectedValues(selectedValues);
+  const handleBlur = () => {
+    if (shouldCreate?.(searchValue, selectedValues)) {
+      setSelectedValues([...selectedValues, searchValue]);
+    }
     setSearchValue("");
-    onBlur?.(event);
   };
 
   const handleSearchChange = (newSearchValue: string) => {
-    setSearchValue(newSearchValue);
+    const values = parseValues(newSearchValue);
+    const validValues = values.filter(value =>
+      shouldCreate?.(value, selectedValues),
+    );
 
-    const isValid = shouldCreate?.(newSearchValue, selectedValues);
-    if (isValid) {
-      setSelectedValues([...lastSelectedValues, newSearchValue]);
-    } else {
-      setSelectedValues(lastSelectedValues);
+    if (values.length > 1) {
+      if (validValues.length > 0) {
+        setSelectedValues([...selectedValues, ...validValues]);
+        setSearchValue("");
+      }
+      return;
     }
+
+    setSearchValue(newSearchValue);
   };
 
   const handlePaste = (event: ClipboardEvent<HTMLInputElement>) => {
-    const text = event.clipboardData.getData("Text");
-    const values = text.split(/[\n,]/g);
-    if (values.length > 1) {
-      const validValues = [...new Set(values)]
-        .map(value => value.trim())
-        .filter(value => shouldCreate?.(value, selectedValues));
-      if (validValues.length > 0) {
-        event.preventDefault();
-        const newSelectedValues = [...lastSelectedValues, ...validValues];
-        setSelectedValues(newSelectedValues);
-        setLastSelectedValues(newSelectedValues);
-      }
+    const text = event.clipboardData.getData("text");
```

Comment:
Using `Text` was breaking the test, and both `Text` and `text` work in my browsers.